### PR TITLE
DOC Performance improvements for changelog

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/developer-docs/issues/330

Performance measurements
- [Eagerloading](https://github.com/silverstripe/developer-docs/issues/330#issuecomment-1689246562)
- [RelationList::foreignID()](https://github.com/silverstripe/silverstripe-framework/pull/10904#issue-1838704020)
- [DataList::byIDs()](https://github.com/silverstripe/silverstripe-framework/issues/10860#issue-1790646019)
- [SiteTree search](https://github.com/silverstripe/silverstripe-elemental/pull/1069)

### CODE REVIEW

File: en/04_Changelogs/5.1.0.md [hunk 1]
```diff
@@ -108,6 +114,13 @@ SilverStripe\ORM\DataList:
   use_placeholders_for_integer_ids: true
 ```
 
+The following performance improvements were measured in a test setup with 10,000 records:
+- DataList::byIDs() - 198% faster - (0.0608s vs 0.1812s)
```
Comment (ChloeSartorelli) [inline en/04_Changelogs/5.1.0.md:118]: How many IDs were being passed in? That is likely a factor.

File: en/04_Changelogs/5.1.0.md [hunk 2]
```diff
@@ -108,6 +114,13 @@ SilverStripe\ORM\DataList:
...
+- RelationList::foreignIDFilter()
+  - HasManyList::foreignIDFilter() - 108% faster (0.1584s vs 0.3304s)
+  - ManyManyList::foreignIDFilter() - 108% faster (0.1529s vs 0.3119s)
+  - ManyManyThroughList::foreignIDFilter() - 27% faster (0.6901s vs (0.8766s)
```
Comment (ChloeSartorelli) [inline en/04_Changelogs/5.1.0.md:122]: How did you test this? I'm guessing you didn't just call `foreignIDFilter()`?

**Review by GuySartorelli — COMMENTED** [review]
Note: I'm not doing a full review, this is just some questions I had in passing.

File: en/04_Changelogs/5.1.0.md (hunk 1 above)
Comment (emteknetnz) [reply en/04_Changelogs/5.1.0.md:118]: 10K - I've linked to the test setup in the description - https://github.com/silverstripe/silverstripe-framework/issues/10860#issue-1790646019

File: en/04_Changelogs/5.1.0.md (hunk 2 above)
Comment (emteknetnz) [reply en/04_Changelogs/5.1.0.md:122]: I've linked to the test setup in the description - https://github.com/silverstripe/silverstripe-framework/pull/10904#issue-1838704020

File: en/04_Changelogs/5.1.0.md (hunk 1 above)
Comment (ChloeSartorelli) [reply en/04_Changelogs/5.1.0.md:118]: Ahh okay. The way this reads to me right now is that there were 10,000 records in the database and an unspecified number being passed into the method. Might be worth explicitly stating that the 10,000 IDs were being passed into the method.

File: en/04_Changelogs/5.1.0.md (hunk 2 above)
Comment (ChloeSartorelli) [reply en/04_Changelogs/5.1.0.md:122]: Might be worth being explicit about the test, something like "This was tested by passing {x number} IDs into the `forForeignID()` method of each list class, then executing the query"

File: en/04_Changelogs/5.1.0.md (hunk 1 above)
Comment (emteknetnz) [reply en/04_Changelogs/5.1.0.md:118]: Updated

File: en/04_Changelogs/5.1.0.md (hunk 2 above)
Comment (emteknetnz) [reply en/04_Changelogs/5.1.0.md:122]: Updated, I omitted the 'then executing the query' bit cos I'm "sharing" the text with byIDs() - however I don't think we need to be that pedantic here, I think the message of "made the method run faster" is clear

**Review by maxime-rainville — CHANGES_REQUESTED** [review]
Maybe this is just me, but I'm not a big fan of the "X% faster" metric in this context. I think its difficult to understand what we are talking about.

I would rather say that the "response time has been reduced by X%".

**Comment by emteknetnz** [comment]
As discussed offline, we'll keep the % faster language since it's more commonly used and easier to say and easiest to understand that it's a "good thing", though it's less "grammatically correct".  Saying that query time has been reduced could be misinterpreted as being a "bad" thing since the word "reduced" is generally negative. Using the word "faster" make it more immediately obvious that it's positive.

e.g.

DataList::byIDs() - 198% faster - (0.0608s vs 0.1812s)

vs

DataList::byIDs() - query time has been reduced by 66% - (0.0608s vs 0.1812s)

**Review by maxime-rainville — APPROVED** [review]

**Pull request merged**